runtime/vm/statistics.c

```diff
@@ -25,23 +25,25 @@
 
 
 void *
-addStatistic (J9JavaVM* javaVM, U_8 * name, U_8 dataType)
+addStatistic(J9JavaVM *javaVM, U_8 *name, U_8 dataType)
 {
-	PORT_ACCESS_FROM_JAVAVM(javaVM);
-
-	IDATA stringSize = strlen((char*)name) + 1;
-	IDATA size = stringSize;
-	J9Statistic * statistic;
-
-	/* Adjust for the default size in the struct array of 4 */
-	size += offsetof(J9Statistic, name);
+	J9Statistic *statistic = NULL;
 
+	PORT_ACCESS_FROM_JAVAVM(javaVM);
 	/* Allocate and fill the structure */
-	statistic = j9mem_allocate_memory(size, OMRMEM_CATEGORY_VM);
-	if (statistic != NULL) {
+	statistic = j9mem_allocate_memory(sizeof(J9Statistic), OMRMEM_CATEGORY_VM);
+	if (NULL != statistic) {
+		IDATA stringSize = strlen((char*)name) + 1;
 		statistic->dataSlot = 0;
 		statistic->dataType = (U_32) dataType;
-		strcpy((char*)&(statistic->name), (char*)name);
+
+		statistic->name = j9mem_allocate_memory(stringSize, OMRMEM_CATEGORY_VM);
```

Comment:
The statistics kept the full data in a single allocation.  This doubles the allocations for a statistic.  It looks like the right amount of memory has been allocated originally....

Any insight into why this is aborting?  Are we missing a "no-strict-aliasing" style compile option here?

Reply:
i think the `&` may have been causing issues. i reverted to the old implementation. but, i changed how the size of `J9Statistic` is evaluated and removed the `&` during `strcpy`. now, the code works on OSX. remaining changes are done to satisfy coding standards.